main/src/main/scala/sbt/Defaults.scala

```diff
@@ -361,14 +361,19 @@ object Defaults extends BuildCommon {
       val include = (includeFilter in unmanagedSources).value
       val exclude = (excludeFilter in unmanagedSources).value match {
         case e =>
-          (managedSources in ThisScope).value match {
-            case l if l.nonEmpty =>
-              e || new FileFilter {
-                private val files = l.toSet
-                override def accept(pathname: File): Boolean = files.contains(pathname)
-                override def toString = s"ManagedSourcesFilter($files)"
-              }
-            case _ => e
+          val s = state.value
+          try {
+            Project.extract(s).runTask(managedSources in Compile in ThisScope, s) match {
```

Comment:
Out of graph execution?  Ouch.

Reply:
Yeah, this whole thing is a mistake. I don't think we should support managedSources writing into the unmanaged source directories. You deserve an infinite loop if you set up your build that way.